### DISCUSSION TITLE
lxqt-panel sample config file can be provided and how to be loaded?

### DISCUSSION THREAD

**Discussion by estatistics** [lxqt/lxqt#2389] [General]
lxqt-panel sample config file can be provided and how to be loaded?

lets say i would like to stick on the lxpanel the terminal, gimp, firefox in a permament base.
How I can do this? 
Also, in .config/lxpanel has 3 folders with panels 
default, lubuntu, lxde and panel. 
Inside panel config, it says to not change it manually. why?
what is launchtaskbar.cfg?

lxpanel has an option -p to load custom config. 
But trying to run it with already produced panel configuration files in those folders failed. why?
What is a proper lxpanel configuration file? You havent provide a single example in whole internet. 
neither in man page or elsewhere. 

So, i am asking how i can write a configuration file for lxpanel, and run it permantly with that...

**Comment by isf63** [comment]
Are you trying to configure lxpanel or lxqt-panel?

`~/.config/lxpanel/` is the directory for LXDE's panel (not LXQt's). The file for lxqt-panel is `~/.config/lxqt/panel.conf`. The panel is configured under [panel\<id\>] and individual plugins under [\<plugin name\>].

If you run `lxqt-panel --help` it shows `-c <Configuration file>` is how to use an alternate config file.

**Reply by stefonarch** [reply]
Default configuration for the first panel is  in ` /usr/share/lxqt/panel.conf`

**Reply by estatistics** [reply]
I cant understand how i can stick terminal on the panel. I can see in the bottom panel that pcmanfm is stick on the panel. But in this panel config of `/usr/share/lxqt/panel.conf` no pcmanfm is showed. it seems that it contains only the plugins. please write a proper guide for lxpanel. it is very messy for someone to understand how this can work.  I hadnt that on kde panels.

panels=panel1

[panel1]
plugins=mainmenu,desktopswitch,quicklaunch,taskbar,statusnotifier,tray,mount,volume,worldclock,showdesktop
position=Bottom
desktop=0

[mainmenu]
type=mainmenu
filterClear=true

[desktopswitch]
type=desktopswitch

[quicklaunch]
type=quicklaunch
alignment=Left

[taskbar]
type=taskbar
buttonWidth=220
closeOnMiddleClick=true
groupingEnabled=false

[mount]
type=mount

[worldclock]
type=worldclock

[volume]
device=0
type=volume

[showdesktop]
alignment=Right
type=showdesktop

[statusnotifier]
alignment=Right
type=statusnotifier

[tray]
type=tray

**Reply by estatistics** [reply]
I can see in that "/usr/share/lxqt/lxqt-panel/" contains desktop files. So I must copy apps that i want to include in this folder as "desktop files" ?

**Reply by estatistics** [reply]
i copied terminal.desktop inside "/usr/share/lxqt/lxqt-panel/" but it is not showed on the widget menu.

**Reply by estatistics** [reply]
Where i must copy .desktop files? I copied in /usr/share/lxqt/lxqt-panel/ term.desktop and running in ~/.config/lxqt/, `lxqt-panel -c panel.conf' says not found term pluggin. 

[term]
alignment=Left
type=term

and... 

plugins=mainmenu, term, quicklaunch, taskbar, mount, volume, tray, statusnotifier, worldclock

**Comment by tsujan** [comment]
@estatistics 
Please don't turn this page into a personal notebook. Stop this!

**Comment by estatistics** [comment]
I deleted my previous comments. Finally, i found how to add  quicklaunch menus. A proper guide will be most helpful than people trying to guess what to do. I expect a proper man page for that.

Finally i did it. 1-2-3 must be set, depending on the number of quicklaunch apps. apps size must follows the numbers of apps. Note that first is taking into consideration the panel.conf in `/home/elias/.config/lxqt/`

I run using it `lxqt-panel -c panel.conf`.
Two places exists as you said about panel.conf: `/usr/share/lxqt/` and `/home/elias/.config/lxqt/`

[quicklaunch]
type=quicklaunch
alignment=Left
apps\1\desktop=/home/elias/Desktop/systemmonitor.desktop
apps\2\desktop=/home/elias/Desktop/operadev.desktop
apps\3\desktop=/usr/share/applications/gimp.desktop
apps\size=3

An app can be awful if not a proper guide exists. Thats true for lxqt-panel.

**Comment by tsujan** [comment]
As it has been said repeatedly, and for several reasons, the config files of GUI apps aren't for manual editing.

To make your desired config file, you should set up the app as you like, exit it, and copy the config file that it has created. If you want to do more, you'll need either to consult the source or rely on trial-and-error.

**Reply by estatistics** [reply]
but lxpanel has not any option in gui to add quicklaunch menus or any other app-basing editing!!!!! Nothing, null, zero.

**Reply by tsujan** [reply]
We don't have `lxpanel` here.

`lxqt-panel` has a config dialog, and each panel plugin has its own config dialog if needed. The user isn't supposed to do any manual editing.

**Reply by tsujan** [reply]
As for Quick Launch, it works with drag-and-drop.

**Reply by estatistics** [reply]
yes, lxqt panel works with drag&drop, but because existed only one icon, it was hard to find. Now that i have added more icons, it works. nevertheless, for any case, a proper guidance must be written.   You said that we can run our manual panel.conf with `lxqt-panel -c panel.conf` option. How i can understand how to construct it when no written guidance exists?

**Reply by tsujan** [reply]
Like with any other app, you can check it in a terminal: `lxqt-panel --help`

**Reply by tsujan** [reply]
> How i can understand how to construct it when no written guidance exists?

You don't construct it; the app does. See https://github.com/lxqt/lxqt/discussions/2389#discussioncomment-5252789
